web/src/ApplicationEditPage.js

```diff
@@ -119,14 +119,17 @@ class ApplicationEditPage extends React.Component {
   getApplication() {
     ApplicationBackend.getApplication("admin", this.state.applicationName)
       .then((application) => {
-        if (application.grantTypes === null || application.grantTypes === undefined || application.grantTypes.length === 0) {
-          application.grantTypes = ["authorization_code"];
+        if (application === undefined || application === null) {
+          this.props.history.push("/404");
```

Comment:
Don't change URL, just show 404 under the current URL like: https://door.casdoor.com/aaa

![image](https://github.com/casdoor/casdoor/assets/3787410/033071a9-b0ca-4546-acff-71c721f3d1a6)